### PR TITLE
EM4100 read/clone/simulate standalone mode

### PR DESCRIPTION
**Pull request by temskiy**
V1 - can only simulate predefined IDs
V2 - read (for RDV4 can store to flash) and simulate prdefined or readed IDs
V3 - see V2 + clone to T5555 tag

### CODE REVIEW

**Comment by iceman1001** [comment]
Nice!

Let's see there are some issues here.
- v2,v3  seem to write direct to flash mem. 
This isn't good since the first 128kb is used for the SPI-FFS implemementation.
I recommened you either use the create/write/append/read file posibilities.

Last run  in order to make sure your code follows nice whitespace / intends.
- make style

**Comment by iceman1001** [comment]
Instead of so many diferent version of your standalone mode,  
how about you add a default file for reading id to simulate to flash /  covers v1.
the v2/v3  seem also being possible to one version.

**Comment by temskiy** [comment]
Thanks for ideas. I will try to make it soon.

**Comment by iceman1001** [comment]
How is it going?

**Comment by temskiy** [comment]
I make V1 more easy. 
Advanced mode (ex. V3) store readed IDs to file in spiffs as you adviced.
And "make style" at the end.

File: armsrc/Standalone/Makefile.inc [hunk 1]
```diff
@@ -33,11 +33,11 @@ endif
...
-ifneq (,$(findstring WITH_STANDALONE_HF_14ASNIFF,$(APP_CFLAGS)))
-    SRC_STANDALONE = hf_14asniff.c
+# WITH_STANDALONE_LF_EM4100EMUL
+ifneq (,$(findstring WITH_STANDALONE_LF_EM4100EMUL,$(APP_CFLAGS)))
+    SRC_STANDALONE = lf_em4100emul.c
 endif
-# WITH_STANDALONE_LF_ICEHID
-ifneq (,$(findstring WITH_STANDALONE_LF_ICEHID,$(APP_CFLAGS)))
-    SRC_STANDALONE = lf_icehid.c
+# WITH_STANDALONE_LF_EM4100RWC
```
Comment (iceman1001) [inline armsrc/Standalone/Makefile.inc]: You are replacing a standalonemode here.   I suggest you add your own after the last

File: armsrc/Standalone/Makefile.hal [hunk 2]
```diff
@@ -35,18 +35,19 @@ define KNOWN_STANDALONE_DEFINITIONS
...
+
 endef
 
-STANDALONE_MODES := LF_SAMYRUN LF_ICERUN LF_PROXBRUTE LF_HIDBRUTE LF_ICEHID
-STANDALONE_MODES += HF_YOUNG HF_MATTYRUN HF_COLIN HF_BOG HF_14ASNIFF
+STANDALONE_MODES := LF_SAMYRUN LF_ICERUN LF_PROXBRUTE LF_HIDBRUTE LF_EM4100EMUL LF_EM4100RWC
+STANDALONE_MODES += HF_YOUNG HF_MATTYRUN HF_COLIN HF_BOG
 STANDALONE_MODES_REQ_SMARTCARD :=
-STANDALONE_MODES_REQ_FLASH := HF_COLIN HF_BOG HF_14ASNIFF LF_ICEHID
+STANDALONE_MODES_REQ_FLASH := HF_COLIN HF_BOG
```
Comment (iceman1001) [inline armsrc/Standalone/Makefile.hal]: why are your removing these ones?

File: armsrc/Standalone/Makefile.hal [hunk 3]
```diff
@@ -35,18 +35,19 @@ define KNOWN_STANDALONE_DEFINITIONS
...
+| LF_EM4100RWC    | Read/simulate em4100 tags & clone it   |
+|                 | to T555x tags                          |
 +----------------------------------------------------------+
+
 endef
 
-STANDALONE_MODES := LF_SAMYRUN LF_ICERUN LF_PROXBRUTE LF_HIDBRUTE LF_ICEHID
-STANDALONE_MODES += HF_YOUNG HF_MATTYRUN HF_COLIN HF_BOG HF_14ASNIFF
+STANDALONE_MODES := LF_SAMYRUN LF_ICERUN LF_PROXBRUTE LF_HIDBRUTE LF_EM4100EMUL LF_EM4100RWC
+STANDALONE_MODES += HF_YOUNG HF_MATTYRUN HF_COLIN HF_BOG
```
Comment (iceman1001) [inline armsrc/Standalone/Makefile.hal]: why are you removing SNIFF/HID modes here?

File: armsrc/Standalone/Makefile.hal [hunk 4]
```diff
@@ -35,18 +35,19 @@ define KNOWN_STANDALONE_DEFINITIONS
...
 +----------------------------------------------------------+
-| HF_14ASNIFF     | 14a sniff to flashmem                  |
-| (RDV4 only)     |                                        |
+| LF_EM4100EMUL   | Simulate predefined em4100 tags only   |
+|                 |                                        |
 +----------------------------------------------------------+
-| LF_ICEHID       | LF HID collector to flashmem           |
-| (RDV4 only)     |                                        |
+| LF_EM4100RWC    | Read/simulate em4100 tags & clone it   |
+|                 | to T555x tags                          |
```
Comment (iceman1001) [inline armsrc/Standalone/Makefile.hal]: same here,  you are replacing instead of  adding your own.

**Comment by iceman1001** [comment]
I added some comments in your two Makefile changes.   You seem to replace instead of adding you own.

**Comment by temskiy** [comment]
Sorry, a forget pull last changes from master. Will make it right now.

**Comment by iceman1001** [comment]
don't forget to add a line in CHANGELOG,md

**Pull request merged**